test/onnx/test_op_consistancy.py

```diff
@@ -0,0 +1,365 @@
+        )
+        decorators.append(new_decorator)
+        opinfo.decorators = tuple(decorators)
+
+    # This decorator doesn't modify fn in any way
+    def wrapped(fn):
+        return fn
+
+    return wrapped
+
+
+def opsets_before(opset: int) -> Callable[[int], bool]:
+    """Returns a comparison function that decides if the given opset is before the specified."""
+
+    def compare(other_opset: int):
+        return other_opset < opset
+
+    return compare
+
+
+def opsets_after(opset: int) -> Callable[[int], bool]:
+    """Returns a comparison function that decides if the given opset is after the specified."""
+
+    def compare(other_opset: int):
+        return other_opset > opset
+
+    return compare
+
+
+# Modify this section ###
+# NOTE: Modify this section as more ops are supported. The list should be sorted
+# alphabetically.
+
+# Ops to be tested for consistency between onnx and pytorch
+ALLOWLIST_OP = (
+    "ceil",
+    "sqrt",
+    "t",
+)
+
+# fmt: off
+# Turn off black formatting to keep the list compact
+
+# Expected failures for onnx export. If an op is expected to fail only for certain
+# ONNX opsets, add the op to EXPECTED_OPSET_FAILS below.
+# The list should be sorted alphabetically by op name.
+EXPECTED_SKIPS_OR_FAILS: Tuple[DecorateMeta, ...] = (
+    skip("ceil", dtypes=BOOL_TYPES + INT_TYPES + QINT_TYPES + COMPLEX_TYPES, reason="not supported by onnx"),
+    skip("ceil", dtypes=[torch.float64], reason="Ceil on f64 not supported by ONNX Runtime"),
+    skip("sqrt", dtypes=BOOL_TYPES + QINT_TYPES + COMPLEX_TYPES, reason="not supported by onnx"),
+    xfail("t", dtypes=COMPLEX_TYPES, reason="jit tracer error for complex types"),
+    skip("t", dtypes=[torch.float16], reason="flaky tests in CI"),
+)
+# fmt: on
+
+# Expected opset specific fails for ops that do not support specific opsets
+
+EXPECTED_OPSET_FAILS: Tuple[XfailOpset, ...] = (
+    # TODO: sqrt for torch.bfloat16 is just an example. Replace it with more meaningful
+    # skips when there are.
+    XfailOpset(
+        "sqrt",
+        dtypes=[torch.bfloat16],
+        opsets=[opsets_before(13)],
+        reason="Sqrt not defined for bf16 before opset 13",
+    ),
+)
+
+# END OF SECTION TO MODIFY ###
+
+
+OPS_DB = copy.deepcopy(common_methods_invocations.op_db)
+
+
+class SingleOpModel(torch.nn.Module):
+    """Test model to wrap around a single op for export."""
+
+    def __init__(self, op, kwargs):
+        super().__init__()
+        self.operator = op
+        self.kwargs = kwargs
+
+    def forward(self, *args):
+        return self.operator(*args, **self.kwargs)
+
+
+class TestConsistency(common_utils.TestCase):
+    """Test consistency of exported ONNX models.
+
+    This is a parameterized test suite.
+    """
+
+    @common_device_type.ops(OPS_DB, allowed_dtypes=SUPPORTED_DTYPES)
+    @skip_ops(
+        OPS_DB,
+        "TestConsistency",
+        "test_output_match",
+        to_skip=EXPECTED_SKIPS_OR_FAILS,
```

Comment:
hmm ... does `xfail` in `EXPECTED_SKIPS_OR_FAILS` expected to fail or instead skipped now that they are passed to `to_skip`

Reply:
They still expect failures. I just copied the decorator and fitted it for our own use. I will change the param name to make it more precise

Reply:
Done